WalletWasabi/Interfaces/EventSourcing/IEventRepository.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.EventSourcing;
+using WalletWasabi.Exceptions;
+
+namespace WalletWasabi.Interfaces.EventSourcing
+{
+	public interface IEventRepository
+	{
+		/// <summary>
+		/// Atomically persistently appends ordered list of events for given aggregate.
+		/// In case of duplicate of <seealso cref="WrappedEvent.SequenceId"/>
+		/// <seealso cref="OptimisticConcurrencyException"/> is thrown indicating that
+		/// command should be retried with freshly loaded events from <see cref="ListEventsAsync"/>.
+		/// </summary>
+		/// <param name="aggregateType">Type of aggregate</param>
+		/// <param name="aggregateId">Id of aggregate</param>
+		/// <param name="wrappedEvents">Ordered list of events to be persisted</param>
+		/// <exception cref="OptimisticConcurrencyException">If there is concurrency conflict ; retry command</exception>
+		/// <exception cref="TransientException">Transient infrastracture failure</exception>
+		/// <exception cref="ArgumentException">Invalid input</exception>
+		public Task AppendEventsAsync(
+			string aggregateType,
+			string aggregateId,
+			IEnumerable<WrappedEvent> wrappedEvents);
```

Comment:
the Nazy bot complains about such a construction. He likes only one line or each argument on a separate line

Reply:
The bot is right. Further than that, I prefer one-line solutions except it is too long and harass readability.

Reply:
my solution is good by the robot. kiminuo's solution is wrong by the robot.

You need to define number what is too long. This is 122 characters. That's too long for me. My limit is 100. So I'm keeping it because "it is too long" and because kimuo's solution is not liked by the robot.